src/main/java/com/synopsys/integration/alert/web/config/JobConfigActions.java

```diff
@@ -172,20 +174,28 @@ public JobFieldModel updateJob(UUID id, JobFieldModel jobFieldModel) throws Aler
         return savedJobFieldModel;
     }
 
-    private void validateJobNameUnique(JobFieldModel jobFieldModel) throws AlertFieldException {
+    private void validateJobNameUnique(@Nullable UUID currentJobId, JobFieldModel jobFieldModel) throws AlertFieldException {
         for (FieldModel fieldModel : jobFieldModel.getFieldModels()) {
-            validateJobNameUnique(fieldModel);
+            validateJobNameUnique(currentJobId, fieldModel);
         }
     }
 
-    private void validateJobNameUnique(FieldModel fieldModel) throws AlertFieldException {
+    private void validateJobNameUnique(@Nullable UUID currentJobId, FieldModel fieldModel) throws AlertFieldException {
         Optional<FieldValueModel> jobNameFieldOptional = fieldModel.getFieldValueModel(ChannelDistributionUIConfig.KEY_NAME);
         String error = "";
         if (jobNameFieldOptional.isPresent()) {
             String jobName = jobNameFieldOptional.get().getValue().orElse(null);
             if (StringUtils.isNotBlank(jobName)) {
                 List<ConfigurationJobModel> jobs = configurationAccessor.getAllJobs();
+
                 boolean foundDuplicateName = jobs.stream()
+                                                 .filter(job -> {
+                                                     if (null != currentJobId) {
+                                                         return !job.getJobId().equals(currentJobId);
+                                                     } else {
+                                                         return true;
+                                                     }
+                                                 })
```

Comment:
Either abstract this into its own method, or do the check before this stream. This is a little to cluttered IMO.